gapic-generator/lib/gapic/schema/api.rb

```diff
@@ -171,16 +187,53 @@ def incode_samples
       #   A Hash of the configuration values.
       def configuration
         @configuration ||= begin
-          config = {}
-
-          if protoc_options[:configuration]
-            config = YAML.load_file protoc_options[:configuration]
-            config.merge! config
+          config_file = protoc_options["configuration"]
+          config = config_file ? YAML.load_file(config_file) : {}
+          protoc_options.each do |k, v|
+            next if k == "configuration"
+            branch = key_to_str(k).split(".").reverse.inject(v) { |m, s| { str_to_key(s) => m } }
+            config = deep_merge config, branch
           end
-
           config
         end
       end
+
+      private
+
+      def parse_parameter str
```

Comment:
Personally I think I might move this to somewhere that it can be unit tested in isolation (and add some tests), mainly just for documentation purposes.

Reply:
I agree, and I did consider breaking out all the `configuration` and `protoc_options` related logic into a separate class. But I kind of wanted to keep this change as targeted as possible for now, as this is a sprint toward getting integrated with synthtool. I do think we should have it on our list to circle back and refactor later.